ledger-core/application/testwalletapi/statemachine/statemachine.go

```diff
@@ -94,7 +89,19 @@ func (s *SMTestAPICall) stepRegisterBargeIn(ctx smachine.ExecutionContext) smach
 	if !ctx.PublishGlobalAliasAndBargeIn(outgoingRef, bargeInCallback) {
 		return ctx.Error(errors.New("failed to publish bargeInCallback"))
 	}
-	return ctx.Jump(s.stepSendRequest)
+	s.registeredBargeInRef = outgoingRef
```

Comment:
to pass a one-time param you can use a closure and do not pollute SM's body, e.g.

outgoingRef := ...
return ctx.Jump(func (ctx) ... {
   // use outgoingRef
})